blue-web/src/main/resources/webapp/js/paper/directives/TocDirective.js

```diff
@@ -62,12 +64,23 @@ angular.module('bluelatex.Paper.Directives.Toc', [])
           }
 
           currentlevel = line.level;
+          if(!numbers[currentlevel]) {
+            numbers[currentlevel] = 1;
+          } else {
+            numbers[currentlevel] ++;
+          }
+          var lineNumber = "";
+          for (var j = 1; j <= currentlevel; j++) {
+            if(numbers[j]) {
+              lineNumber += numbers[j] + ".";
+            }
+          }
```

Comment:
Because the variable ```numbers``` is not an array but an ```object``` and because I stop the loop at ```currentLevel``` and I don't iterate over the complete object.